enumerable_methods.rb

```diff
@@ -0,0 +1,106 @@
+# Project 2: Enumerable Methods
+module Enumerable
+  def my_each(&block)
+    size.times { |i| block.call(self[i]) } and return self if block
+    to_enum
+  end
+
+  def my_each_with_index(&block)
+    my_each { |idx| block.call(self[idx], idx) } and return self if block_given?
+    to_enum
+  end
+
+  def my_select(&block)
+    selection = []
+    my_each { |item| selection << item if block.call(item) } and return selection if block_given?
+    to_enum
+  end
+
+  def my_map
+    mapped = []
+    my_each { |item| mapped << yield(item) if yield(item) != 0 } and return mapped if block_given?
+    to_enum
+  end
+
+  def my_all?(args = nil, &block)
+    swap = true
+    if block
+      my_each { |item| swap = false unless block.call(item) }
+    elsif args.nil?
+      my_each { |item| swap = false unless self[item] }
+    else
+      my_each { |item| swap = false unless args === self[item] }
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true when none of the collection members are false or nil
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex